lib/bokken/documents.ex

```diff
@@ -86,10 +86,26 @@ defmodule Bokken.Documents do
       {:error, %Ecto.Changeset{}}
 
   """
+  alias Bokken.Accounts.User
+
   def create_file(attrs \\ %{}) do
-    %File{}
-    |> File.changeset(attrs)
-    |> Repo.insert()
+    max_total_size = String.to_integer(System.get_env("MAX_TOTAL_SIZE", "6000000"))
+
+    if verify_total_size(attrs.document, attrs.user_id) > max_total_size do
+      {:error, "You exceeded the maximum storage quota. Try to delete one or more files"}
+    else
+      %File{}
+      |> File.changeset(attrs)
+      |> Repo.insert()
+
+      user = Accounts.get_user!(attrs.user_id)
+
+      user
+      |> User.changeset(%{total_file_size: verify_total_size(attrs.document, attrs.user_id)})
+      |> Repo.update()
+
+      {:ok, %File{}}
```

Comment:
1. This does not return a useful result. It is a constant function after all
2. This must be enclosed within a transaction, so that either both operations work or both fail